src/server.rs

```diff
@@ -241,7 +242,9 @@ fn cleanup_acks(
         let min_timestamp = time.elapsed().saturating_sub(mutations_timeout);
         for mut ticks in &mut clients {
             ticks.cleanup_older_mutations(min_timestamp, |mutate_info| {
-                mutate_info.entities.clear();
+                for (_, components) in mutate_info.entities.drain(..) {
+                    pools.components.push(components);
+                }
```

Comment:
Change to method on pools that takes `mutate_info.entities.drain(...)` iterator as input.